src/site/content/en/notifications/codelab-notifications-service-worker/index.md

```diff
@@ -0,0 +1,192 @@
+
+1.  Now that the user can register a service worker, you can get a reference to the service worker registration object. In `public/index.js`, replace the `getRegistration` function with the following code:
+
+    ```js
+    // Get the current service worker registration.
+    function getRegistration() {
+      return navigator.serviceWorker.getRegistration();
+    }
+    ```
+
+    The function above uses the
+    [Service Worker API](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API)
+    to get the current service worker registration, if it exists.
+    It makes getting a reference to the service worker registration a bit more convenient.
+
+*   To complete the service worker registration functionality, add code to unregister the service worker. Replace the `unRegisterServiceWorker` function with the following code:
+
+    ```js
+    // Unregister a service worker, then update the UI.
+    async function unRegisterServiceWorker() {
+      // Get a reference to the service worker registration.
+      let registration = await getRegistration();
+      // Await the outcome of the unregistration attempt
+      // so that the UI update is not superceded by a 
+      // returning Promise.
+      await registration.unregister();
+      updateUI();
+    }
+    ```
+
+In the tab where you're viewing the live app, reload the page. The **Register service worker** and **Unregister service worker** buttons should now be working.
+
+## Send notifications to the service worker
+
+In this step, you'll write code that will run when the user clicks **Send a notification** in the app UI. This code will create a notification, check that a service worker is registered, and then send the notification to the service worker using its `postMessage` method.
+
+In the embedded Glitch editor, open `public/index.js` and
+replace the `sendNotification` function with the following code:
+
+```js
+// Create and send a test notification to the service worker.
+async function sendNotification() {
+  // Use a random number as part of the notification data
+  // (so you can tell the notifications apart during testing!)
+  let randy = Math.floor(Math.random() * 100);
+  let notification = {
+    title: 'Test ' + randy,
+    options: { body: 'Test body ' + randy }
+  };
+  // Get a reference to the service worker registration.
+  let registration = await getRegistration();
+  // Check that the service worker registration exists.
+  if (registration) {
+    // Check that a service worker controller exists before
+    // trying to access the postMessage method.
+    if (navigator.serviceWorker.controller) {
+      navigator.serviceWorker.controller.postMessage(notification);
+    } else {
+      console.log('No service worker controller found. Try a soft reload.');
+    }
+  }
+}
+```
+
+Here's what that code is doing:
```

Comment:
I thought an explicit intro to this list might be helpful. Cut if you don't like it!